### ISSUE TITLE
user defined fp functions do not error if incorrect number of arguments given

### ISSUE THREAD

**Issue by mbertucci47** [latex3/latex3#1639]
A user-defined fp function does not error if given the incorrect number of arguments. For example,
```tex
\documentclass{article}
\begin{document}
\ExplSyntaxOn
\fp_new_function:n { firstoftwo }
\fp_set_function:nnn { firstoftwo } { a , b } { a }
\fp_eval:n { firstoftwo ( 1,2,3 ) }
\fp_eval:n { firstoftwo ( 4 ) }
\ExplSyntaxOff
\end{document}
```
happily prints `14`. I understand this may be expected behavior for speed reasons, but it is different than the behavior for predefined functions so I think deserves to be documented. For example
```tex
\fp_eval:n { ln ( 4,5 ) }
```
errors with
```
! LaTeX Error: ln got more than one argument; used nan.
```

**Comment by mbertucci47** [comment]
Just my opinion, but I'd prefer an error if the wrong number of arguments is used. It makes debugging easier. Also, user-defined fp functions are relatively new, so I don't think too many users will be affected.

**Comment by Skillmon** [comment]
I agree with @mbertucci47, though another interface to define variadic functions would be really nice (for instance `mean()` and `polyval()` come to mind)

**Comment by car222222** [comment]
Agreed, but I would not think of such as a "variadic function"!  Rather, they are functions of one variable, and that variable is an ordered list (or, often, just a set: ie unordered).
